garage/tf/algos/batch_polopt.py

```diff
@@ -66,22 +55,16 @@ def __init__(self,
         self.policy = policy
         self.baseline = baseline
         self.scope = scope
-        self.n_itr = n_itr
-        self.start_itr = start_itr
-        self.batch_size = batch_size
         self.max_path_length = max_path_length
         self.discount = discount
-        self.gae_lambda = gae_lambda
         self.plot = plot
         self.pause_for_plot = pause_for_plot
-        self.center_adv = center_adv
-        self.positive_adv = positive_adv
         self.store_paths = store_paths
-        self.whole_paths = whole_paths
-        self.fixed_horizon = fixed_horizon
+        self.es = exploration_strategy
+        self.input_include_goal = input_include_goal
         if sampler_cls is None:
             if self.policy.vectorized and not force_batch_sampler:
-                sampler_cls = VectorizedSampler
+                sampler_cls = OffPolicyVectorizedSampler if self.off_policy else OnPolicyVectorizedSampler
```

Comment:
@krzentner again to look at the dispatch strategy used here, before it gets out-of-hand

Reply:
We could have a function that builds the sampler that subclasses, i.e.OffPolicyBatchPolopt can override. 
I have doubts about the OffPolicyBatchPolopt class. The `OffPolicyBatchPolopt.train()` is more specific for DDPG and not generalized. It might be better to get rid of OffPolicyBatchPolopt.